staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -106,11 +106,10 @@ type Route struct {
 }
 
 func (f *Cloud) addCall(desc string) {
-	f.addCallLock.Lock()
-	defer f.addCallLock.Unlock()
-
 	time.Sleep(f.RequestDelay)
 
+	f.addCallLock.Lock()
+	defer f.addCallLock.Unlock()
```

Comment:
Good catch!

Reply:
haha, now I think it was fun, but it drove me crazy, adding more workers to the test didn't give any improvement